src/lib/updateGrantsRecipients.js

```diff
@@ -224,6 +225,103 @@ export const updateCDIGrantsWithOldGrantData = async (grantsToUpdate) => {
   }
 };
 
+export async function syncGeoRegionGroups({ transaction }) {
+  const useTransaction = transaction || await sequelize.transaction();
+  const createdHere = !transaction;
+
+  try {
+    logger.info('Starting Geo Region Groups Sync');
+
+    // Get all geo region Groups
+    const regionGroups = await Group.findAll({
+      where: { name: { [Op.in]: REGIONS } },
+      transaction,
+    });
+    const regionMap = new Map(regionGroups.map((g) => [g.name, g.id]));
+    // Get all replacing grants to protect them
+    const replacingGrantIds = (await GrantReplacements.findAll({
+      attributes: ['replacingGrantId'],
+      transaction,
+    })).map((r) => r.replacingGrantId);
+
+    const safeReplacingIds = replacingGrantIds.length > 0 ? replacingGrantIds : [-1];
```

Comment:
Why have an array with -1 when we have an array with no length? Just seems a bit redundant but maybe I haven't found the usage yet.

Reply:
Had same thought, as I would expect this to be an empty array ( or undefined) rather than `[-1]`. Is a special value when seen somewhere ( `safeReplacingIds` is referenced one other time as part of the replacements of the `sequalize` query.
```

      // Remove stale GroupGrants (excluding replacing grants)
      const result = await sequelize.query(

Reply:
Good question! The `[-1]` is a safety hack to avoid SQL errors when using `WHERE ... NOT IN (:ids)`. If we pass an empty array to sequelize replacements, Postgres will throw a syntax error (`NOT IN ()`). Using `[-1]` ensures the query is always valid and will never filter out any real grantId.